spock-core/src/main/java/org/spockframework/mock/EmptyOrDummyResponse.java

```diff
@@ -90,6 +113,7 @@ public Object respond(IMockInvocation invocation) {
     }
 
     if (returnType == OPTIONAL) return ReflectionUtil.invokeMethod(null, optionalEmptyMethod);
+    if (returnType == COMPLETABLE_FUTURE) return ReflectionUtil.invokeMethod(null, completableFutureCompletedFutureMethod, (Object) null);
```

Comment:
I see that in the end you resigned from the non null value returned from the completable future :).